src/lang-go.ts

```diff
@@ -727,6 +733,94 @@ export async function activateUsingWebSockets(ctx: sourcegraph.ExtensionContext)
     ctx.subscriptions.add(panelView)
 }
 
+function activateImprecise(ctx: sourcegraph.ExtensionContext): void {
+    ctx.subscriptions.add(
+        registerFeedbackButton({
+            languageID: 'go',
+            sourcegraph,
+            isPrecise: false,
+        })
+    )
+
+    activateBasicCodeIntel({
+        sourcegraph,
+        languageID: 'go',
+        fileExts: ['go'],
+        filterDefinitions: ({ repo, filePath, pos, fileContent, results }) => {
+            const currentFileImportedPaths = fileContent
+                .split('\n')
+                .map(line => {
+                    // Matches the import at index 3
+                    const match = /^(import |\t)(\w+ |\. )?"(.*)"$/.exec(line)
+                    return match ? match[3] : undefined
+                })
+                .filter((x): x is string => Boolean(x))
+
+            const currentFileImportPath = repo + '/' + path.dirname(filePath)
+
+            const filteredResults = results.filter(result => {
+                const resultImportPath = result.repo + '/' + path.dirname(result.file)
+                return (
+                    currentFileImportedPaths.some(i => resultImportPath.includes(i)) ||
+                    resultImportPath === currentFileImportPath
+                )
+            })
+
+            return filteredResults.length === 0 ? results : filteredResults
+        },
+        commentStyle: {
+            lineRegex: /\/\/\s?/,
+        },
+    })(ctx)
+}
+
+async function activatePreciseUsingLspClient(ctx: sourcegraph.ExtensionContext): Promise<void> {
```

Comment:
This function `activatePreciseUsingLspClient` is really the only new piece in this PR. The other changes were due to restructuring the code.